pkg/adapter/v2/config.go

```diff
@@ -190,6 +191,10 @@ func (e *EnvConfig) GetLeaderElectionConfig() (*kle.ComponentConfig, error) {
 
 func defaultLeaderElectionConfig() *kle.ComponentConfig {
 	return &kle.ComponentConfig{
-		LeaderElect: false,
+		ResourceLock:  "leases",
```

Comment:
For adapters, we might want to use ConfigMap resource locks since that's the favored way to pass CRs to adapters. We don't have to make the change now.

Reply:
In 0.16 this field was narrowed to just leases.  Endpoints and ConfigMaps are supported by the K8s library largely for (much) earlier versions of K8s than we support, which didn't yet have this API.